driver/controller.go

```diff
@@ -127,8 +149,25 @@ func (s controllerService) CreateVolume(ctx context.Context, req *csi.CreateVolu
 	}
 
 	name = strings.ToLower(name)
+	// If a volume has a source, it has to provisioned on the same node and device class as the source volume.
+	if sourceID != "" {
+		if req.GetAccessibilityRequirements() == nil {
+			node = sourceVol.Spec.NodeName
+		} else {
+			if node != sourceVol.Spec.NodeName {
+				return nil, status.Errorf(codes.InvalidArgument,
+					"nodes mismatch. volumes created from source should be provisioned on same node as the source. Node: '%s' , SourceVolumeNode:'%s'",
+					node, sourceVol.Spec.NodeName)
+			}
+		}
 
-	volumeID, err := s.lvService.CreateVolume(ctx, node, deviceClass, name, requestGb)
+		if deviceClass != req.GetParameters()[topolvm.DeviceClassKey] {
```

Comment:
On line 31, 
deviceClass := req.GetParameters()[topolvm.DeviceClassKey]

This will never be false.

Reply:
This should be 
`if deviceClass != sourceVol.Spec.NodeName {`